kubeadm/hack/publish-kubeproxy.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/Masterminds/semver"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+const (
+	supportedVersions = ">= 1.17.0"
+)
+
+func dockerBuild(v *semver.Version, dockerfilePath string) error {
+	cmd := exec.Command("docker", "build", "--pull",
+		fmt.Sprintf("--build-arg=k8sVersion=v%s", v.String()),
+		fmt.Sprintf("--tag=sigwindowstools/kube-proxy:v%s", v.String()),
+		dockerfilePath,
+	)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func dockerPush() error {
+	cmd := exec.Command("docker", "push", "sigwindowstools/kube-proxy")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func main() {
+	dockerfilePath := os.Args[1]
+	constraint, err := semver.NewConstraint(supportedVersions)
```

Comment:
i have not tested this before.
given we don't have unit tests, hoping that we will not trigger a push for "all" kube-proxy images. :)

Reply:
Worst case is still just a whole lot of images get pushed to docker hub, not a doomsday scenario. But it works in my tests, like I mentioned you can see the last run before I rebased and removed the push trigger here: https://github.com/benmoss/sig-windows-tools/actions/runs/53251162